mousestyles/dynamics/__init__.py

```diff
@@ -431,3 +431,98 @@ def get_score(true_day, simulated_day, weight=[1, 10, 50, 1]):
+        that it optimizes on
+
+    Returns
+    -------
+    best_interval_length: int
+        a interger indicating the optimized time interval
+        in terms of the evaluation score
+    best_fake_mouse: list
+        a list of integers indicating the simulated
+        states using the best_interval_length
+    best_score: float
+        a float representing the evaluation score
+        comparing the best_fake_mouse with the real mice
+        behavior in the strain under the same optimized
+        time interval length.
+
+    Examples
+    --------
+    >>> row_i = np.hstack((np.zeros(40)))
+    >>> ime_df_eg = np.vstack((row_i, row_i, row_i))
+    >>> time_df_eg = pd.DataFrame(time_df_eg)
+    >>> time_df_eg.rename(columns={0:'strain'}, inplace=True)
+    >>> find_best_interval(time_df_eg, 0, np.arange(10, 40, 10))
+    >>> (10, array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                    0, 0]), 0.98245614035087725)
+"""
+    # check all the inputs
+    condition_df = (type(df) == pd.core.frame.DataFrame)
+    condition_strain_num = (strain_num in (0, 1, 2))
+    condition_interval_length_initial = (type(interval_length_initial) ==
+                                         np.ndarray and
+                                         np.sum(interval_length_initial > 0) ==
+                                         len(interval_length_initial) and
+                                         all(isinstance(i, np.int64) for i
+                                             in interval_length_initial))
+
+    if not condition_df:
+        raise ValueError("df should be pandas DataFrame")
+    if not condition_strain_num:
+        raise ValueError("strain_num can only be 0, 1, 2")
+    if not condition_interval_length_initial:
+        raise ValueError("interval_length_initial positive np.array")
+
+    interval_length_initial = [int(interval_length) for interval_length in
+                               interval_length_initial]
+    data_strain = df[df.strain == strain_num]
+    time_score = []
+    store_fake_mouse = []
+    for interval_length in interval_length_initial:
+        prob_matrix_list = get_prob_matrix_list(data_strain, interval_length)
+        np.random.seed([0])
+        fake_mouse = mcmc_simulation(prob_matrix_list, interval_length)
+        store_fake_mouse.append(fake_mouse)
+        real_mouse = data_strain.ix[:, 3:]
+        real_mouse = real_mouse.reset_index()
+        n_mouse = len(real_mouse)
+        mouse_score = np.zeros(n_mouse)
+        for mouse_index in range(0, n_mouse):
+            real_mouse_list = list(real_mouse.ix[mouse_index, :])
+            mouse_score[mouse_index] = get_score(np.array(real_mouse_list),
+                                                 fake_mouse)
+        time_score.append(np.mean(mouse_score))
+        # print(interval_length, np.mean(mouse_score))
```

Comment:
I will delete the code since I used it for checking the progress and don't need it anymore.